apps/frontend/src/components/auth/RequireAuth.tsx

```diff
@@ -0,0 +1,21 @@
+import { useAuthenticator } from '@aws-amplify/ui-react';
+import { Box, Button, Heading } from '@chakra-ui/react';
+import { Navigate, useLocation } from 'react-router-dom';
+
+export function RequireAuth({ children }: { children: JSX.Element }) {
+  const location = useLocation();
+  const { route } = useAuthenticator((context) => [context.route]);
+  const { user, signOut } = useAuthenticator((context) => [context.user]);
+  if (route !== 'authenticated') {
+    return <Navigate to="/login" state={{ from: location }} replace />;
+  }
+  return (
+    <>
+      <Box display="flex" padding="8" justifyContent="space-between">
+        <Heading size="l">Logged in as: {user.username}</Heading>
```

Comment:
can you change this to {user.attributes?.email} so it displays their email address instead of the string of characters for username?

Reply:
<img width="671" alt="Screenshot 2023-04-06 at 9 35 03 PM" src="https://user-images.githubusercontent.com/55663537/230522629-f4bcd710-9d08-4a7a-8290-52e6a5eb6cc9.png">

Reply:
ooh yes good catch, just pushed!